### PR TITLE
Do not mark image edges in 'subpixel' mode of find_boundaries

### LINKED ISSUE #5432: 'subpixel' boundaries at the edges of an image
## Description

When using the function `find_boundaries` with `mode='subpixel'` the behavior at the edges of the image seems strange to me.

From the documentation it is not clear to me what the intended behavior should be. Without an option to explicitly specify the behavior at the edges of the image, I would expect a constant input image to have no boundary at all or maybe straight lines at the edges of the image. However, for constant images with a non-zero value the actual image alternates between `True` and `False`. If the image is completely zero, there is no such effect.

## Way to reproduce
```python
import skimage.segmentation
import numpy as np
skimage.segmentation.find_boundaries(np.ones((3,3), dtype=int), mode='subpixel')
```
returns
```python
array([[False,  True, False,  True, False],
       [ True, False, False, False,  True],
       [False, False, False, False, False],
       [ True, False, False, False,  True],
       [False,  True, False,  True, False]])

```

I would expect the same output as `np.zeros((5,5), dtype=bool)`.

## Version information
```python
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print("scikit-image version: {}".format(skimage.__version__))
import numpy; print("numpy version: {}".format(numpy.__version__))
```

```python
3.7.10 | packaged by conda-forge | (default, Feb 19 2021, 16:07:37)
[GCC 9.3.0]
Linux-5.4.0-74-generic-x86_64-with-debian-bullseye-sid
scikit-image version: 0.18.1
numpy version: 1.20.3
```

### PR DESCRIPTION
**Pull request by grlee77**
## Description

This PR closes #5432, following the suggestion by @firerain42. If there is some reason we want to keep the option with the old behavior of treating the boundary of the image as an edge, let me know. At the moment I am considering this a bug fix and so have not set up a deprecation cycle. 

It seems none of the other `mode` options to `find_boundary` mark the edge as a boundary, it was only the subpixel mode. A test was added that demonstrates this for all modes.

@firerain42, if you are interested in being credited as a co-author on the commit message, please provide a name and email address and I will add a comment for that

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Review by jni — APPROVED** [review]

**Review by rfezzani — APPROVED** [review]
Thank you @grlee77, I'll wait for @firerain42 answer if he want to be sited as co-author before merging :wink:

**Comment by firerain42** [comment]
Thanks @grlee77 for implementing the fix and @jni and @rfezzani for the review.

If it's no hassle for you, I would like to be a co-author (Lukas Petrich <firerain42@googlemail.com>). Thanks.

**Comment by grlee77** [comment]
Thanks @firerain42, I added the text for co-authorship while creating the merge commit. You can see it here:
https://github.com/scikit-image/scikit-image/commit/44ff414094392d27dc1a5925292f3ac35b9a1292

**Pull request merged**
